workflows/check_commit_message_for_tickets.yml

```diff
@@ -0,0 +1,92 @@
+name: Check Commit Message
+
+on:
+  pull_request:
+    types: [opened, synchronize, edited, reopened]
+
+jobs:
+  check-commit-message:
+    runs-on: ubuntu-latest
+    env:
+      REGEX_PATTERN: '^(?i)(ref|refs|reference|references|res|resolve|resolves)[ \t]*:[ \t]*(gh-|\#|\!|[A-Za-z]+-)\d+\s*$'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install jq
+        run: sudo apt-get install jq
+
+      - name: Fetch PR data
+        id: pr_data
+        run: |
+          PR_DATA=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+                    "https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}")
+          echo "PR_DATA<<EOF" >> $GITHUB_ENV
+          echo "$PR_DATA" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - name: Check PR body
+        run: |
+          # Extract the PR body from the PR_DATA environment variable
+          PR_BODY=$(echo "$PR_DATA" | jq -r '.body')
+          echo -e "$PR_BODY" > temp_message.txt
+          echo -e "PR Body:\n$PR_BODY"
+          if grep -Pq "$REGEX_PATTERN" temp_message.txt; then
+            echo -e "PR body meets the requirements.\n"
+            exit 0
+          else
+            echo """
+            PR body does not meet the requirements.
+
+            If PR resolves an issue or ticket, amend the PR body to
+            include a reference to the issue or ticket like so:
+            
+            resolves: PROJ-1234
+            or
+            resolves: #123 (if resolving a GitHub issue)
+
+            If PR does not resolve an issue or ticket, but is part of
+            work done on an issue, amend the PR body to include a
+            reference to the issue or ticket like so:
+
+            reference: PROJ-1234
+            or
+            reference: #123 (if referencing a GitHub issue)
+
+            """
+            exit 1
+          fi
+
+
+      - name: Check PR commit messages
+        run: |
+          COMMIT_DATA=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
```

Comment:
Could be done using `gh pr view ${{ github.event.pull_request.number }} --json commits --jq ...`